packages/react-button/src/components/Button/Button.tsx

```diff
@@ -12,8 +11,20 @@ import { renderButton } from './renderButton';
 export const Button = React.forwardRef<HTMLElement, ButtonProps>((props, ref) => {
   const state = useButton(props, ref);
 
-  useButtonClasses(state);
-  useInlineTokens(state, '--button');
+  const receivedChildren = !!state.children?.children;
+  const receivedIcon = !!state.icon?.children;
+
+  const iconOnly = receivedIcon && !receivedChildren;
```

Comment:
What about the `iconOnly` boolean flag specified in types?

Reply:
I'm not sure that I follow this question. There is an iconOnly defined in the matcher, which is created here. Could you expound?

**A little more info**
This `iconOnly` is computed based on props received. If the user passed only an icon, then this value is true.

Since there is an `iconOnly` variant in the design spec, we pass `iconOnly` as a matcher to makeStyles. There, it is used to apply the correct styling for buttons that only have an icon.

Reply:
I meant that there is an `iconOnly` prop in the `types` file. That should either be accounted for in here or removed from the types file. That's what I meant.